src/hivpy/sexual_behaviour.py

```diff
@@ -1,35 +1,67 @@
-        self.sex_behaviour_trans_male = np.random.choice(sb.sex_behaviour_trans_male_options)
-        self.sex_behaviour_trans_female = np.random.choice(sb.sex_behaviour_trans_female_options)
+        self.sex_behaviour_trans = {SexType.Male:
+                                    self.select_matrix(sb.sex_behaviour_trans_male_options),
+                                    SexType.Female:
+                                    self.select_matrix(sb.sex_behaviour_trans_female_options)}
         self.baseline_risk = sb.baseline_risk  # Baseline risk appears to only have one option
-        self.sex_mixing_matrix_female = np.random.choice(sb.sex_mixing_matrix_female_options)
-        self.sex_mixing_matrix_male = np.random.choice(sb.sex_mixing_matrix_male_options)
-        self.short_term_partners_male = np.random.choice(sb.short_term_partners_male_options)
-        self.short_term_partners_female = np.random.choice(sb.short_term_partners_female_options)
+        self.sex_mixing_matrix = {SexType.Male:
+                                  self.select_matrix(sb.sex_mixing_matrix_male_options),
+                                  SexType.Female:
+                                  self.select_matrix(sb.sex_mixing_matrix_female_options)}
+        self.short_term_partners = {SexType.Male:
+                                    self.select_matrix(sb.short_term_partners_male_options),
+                                    SexType.Female:
+                                    self.select_matrix(sb.short_term_partners_female_options)}
+
+    # Haven't been able to locate the probabilities for this yet
+    # Doing them uniform for now
+    def init_sex_behaviour_groups(self, population):
+        population["sex_behaviour"] = np.where(population['sex'] == SexType.Male,
+                                               np.random.choice(MaleSexBehaviour).value,
+                                               np.random.choice(FemaleSexBehaviour).value)
 
     # Here we need to figure out how to vectorise this which is currently blocked
-    # by the gender if statement
-    def prob_transition(self, gender, age, i, j):
+    # by the sex if statement
+    def prob_transition(self, sex, age, i, j):
         """Calculates the probability of transitioning from sexual behaviour
-        group i to group j, based on gender and age."""
-        if(gender == 1):
-            transition_matrix = self.sex_behaviour_trans_female
-            gender_index = 1
-        else:
-            transition_matrix = self.sex_behaviour_trans_male
-            gender_index = 0
+        group i to group j, based on sex and age."""
+        transition_matrix = self.sex_behaviour_trans[sex]
 
-        age_index = min((int(age)-15)//5, 9)
+        age_index = np.minimum((age.astype(int)-15)//5, 9)
 
-        risk_factor = self.baseline_risk[age_index][gender_index]
+        risk_factor = (self.baseline_risk[age_index]).transpose()[sex.value]
```

Comment:
I'd remove the `(...)` as I don't think they make it more readable, but that's just me!
```suggestion
        risk_factor = self.baseline_risk[age_index].transpose()[sex.value]
```
Actually, do we need the `transpose`? Why not index both dimensions at the same time:
```suggestion
        risk_factor = self.baseline_risk[age_index, sex.value]
```

Reply:
And a note for the future: it would be nice to move consistently to using the `SexType` values directly rather than as numerical indices, but that can come when we read in the data from a static file.

Reply:
That is better, I'm not that familiar with this `array[i,j]` notation which I will need to brush up on because I'm used to `array[i][j]` but for 2D arrays, but this doesn't work if your first index is a list of indices!